src/neo/Network/P2P/Payloads/Header.cs

```diff
@@ -217,7 +217,7 @@ public JObject ToJson(ProtocolSettings settings)
             json["previousblockhash"] = prevHash.ToString();
             json["merkleroot"] = merkleRoot.ToString();
             json["time"] = timestamp;
-            json["nonce"] = nonce.ToString("X");
+            json["nonce"] = nonce.ToString("X8");
```

Comment:
Sorry, my bad. It should be 8 bytes which should be 16 digits.

```suggestion
            json["nonce"] = nonce.ToString("X16");
```